clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -1069,7 +1069,10 @@ static bool buildArgTys(ASTContext &Context, CXXRecordDecl *KernelObj,
           continue;
         }
       }
-      if (!ArgTy.isTriviallyCopyableType(Context)) {
+
+      CXXRecordDecl *RD =
+          cast<CXXRecordDecl>(ArgTy->getAs<RecordType>()->getDecl());
+      if (!RD->hasTrivialCopyConstructor()) {
```

Comment:
I'm afraid that we also need to add check on something like `isTriviallyDestructible`, because AFAIK if class has non-trivial destructor, the CodeGen will emit corresponding kernel argument without `byval` attribute and it can cause problems. If I remember correct @s-kanaev recently fixed these problems for stream class here https://github.com/intel/llvm/pull/1069 .

Reply:
Ah, neat!  I was wondering if that was the case.  My first version of this I checked for a trivial destructor as well, but I couldn't come up with a good justification for that.  I'll fix that.